### PR TITLE
Fixes #127 - allows for case insensitive checking of element names

### LINKED ISSUE #127: HTML element names are matched case-sensitively when using XPATH/CSS selectors
Currently Oga matches element/attribute names/values in xpath/css selectors in a case sensitive way.
```ruby
doc = Oga.parse_html('<!DOCTYPE html><DIV></DIV>')
doc.css('DIV') # => NodeSet(Element(name: "DIV"))
doc.css('div') # => NodeSet()
doc.xpath('//div') # => NodeSet()
```

However, in reality case matching of element names is sensitive in XML but is case insensitive in HTML.

See http://www.w3.org/TR/html51/semantics.html#case-sensitivity and http://dev.w3.org/csswg/selectors/#case-sensitive in regard to CSS selectors.

I didn't find a relevant section in specifications about XPath and HTML. Browsers match element names case-insensitively.

Case matching of attribute names and values seems to be a bit more quirky though. Not sure if any change in Oga is needed. Here are some results from Google Chrome when trying to find element
```html
<input type="text" class="cls">
```
```
$$('.cls') => found
$$('.CLS') => not found
$$('INPUT[TYPE=TEXT]') => found
$$('INPUT[CLASS=cls]') => found
$$('INPUT[CLASS=CLS]') => not found
$x("//INPUT[@type='text']") => found
$x("//INPUT[@TYPE='text']") => not found
```

The following links might be worth checking:
http://sourceforge.net/p/htmlunit/bugs/764/
https://www.w3.org/Bugs/Public/show_bug.cgi?id=24005

### PR DESCRIPTION
**Pull request by dfockler**

### CODE REVIEW

File: lib/oga/xpath/evaluator.rb [hunk 1]
```diff
@@ -1691,7 +1691,9 @@ def type_matches?(xml_node, ast_node)
       def name_matches?(xml_node, name)
         return false unless xml_node.respond_to?(:name)
 
-        name == STAR ? true : xml_node.name == name
+        return true if name == STAR
+
+        xml_node.name == name || xml_node.name.downcase == name.downcase
```
Comment (Arcovion) [inline lib/oga/xpath/evaluator.rb]: `xml_node.name.casecmp(name).zero?` is faster/more efficient here

Comment (dfockler) [inline lib/oga/xpath/evaluator.rb]: @Arcovion Thanks, didn't realize that existed.

**Comment by YorickPeterse** [comment]
Thanks! :heart_eyes:

**Comment by YorickPeterse** [comment]
Worth mentioning: since I'm working on replacing the `XPath::Evaluator` class we'll have to re-apply the same changes for the upcoming `XPath::Compiler` class (see branch https://github.com/YorickPeterse/oga/tree/xpath-compiler). Since the branch is a bit of a mess at the moment (the code is far from done) I'll create a separate issue to keep track of this.

**Pull request merged**